### ISSUE TITLE
Feature proposal: Integrated profiling

### ISSUE THREAD

**Issue by csadorf** [glotzerlab/signac-flow#107]
### Feature description

I suggest that we implement an integrated profiling tool to make it easier to optimize the operations-graph and speed up function such as `status`.

### Proposed solution

I suggest that we add a `--profile` option to the `FlowProject` `status` command that executes the status update, but then provides the user with profiling information about which condition functions are responsible for the majority of the runtime.

### Additional context

Even without this feature, users can profile their status function like this:

```
python -m cProfile -s time project.py status --no-parallelize
```

However, it is kind of tedious to need to remember this command and the output is not always intuitive. I'd prefer to analyze the profiling results and provide a more meaningful report on which condition function is responsible for major delay and why.

In addition, the standard profiler will not provide any meaningful result unless the `--no-parallelize` option is provided, however there are profilers such as [yappi](https://github.com/sumerc/yappi) that would work even in the parallelized status compilation.

Finally, we could do some additional "meta"-profiling such as analyzing the number of required `os.stat` calls and their performance and then provide some recommendations on a reasonable data space size for that specific file system.

**Comment by csadorf** [comment]
@justinGilmer Do you think that would be useful?

**Comment by justinGilmer** [comment]
@csadorf Very much so! I normally use tools like [vprof](https://github.com/nvdv/vprof) when I am trying to profile, but having a dedicated profiling option that will parse the information and give a nicer summary will be very welcome!

**Comment by csadorf** [comment]
Great! Concerning vprof: I had looked at various profiling tools, but didn't come across this one yet. Looks really cool! Unfortunately we might not be able to use it for this specific case, because it has npm requirements.

**Comment by justinGilmer** [comment]
Yeah I do not think vprof is the best option here, its quite useful for personal projects though!

I do think that parsing some profiler output and presenting that to the user would be super helpful, im not entirely sure what the best option is for that, however.

**Comment by csadorf** [comment]
Resolved  with merge of #110 .
